pkg/cmd/prerunner.go

```diff
@@ -39,6 +39,7 @@ var wrongLoginCommandsMap = map[string]string{
 type PreRunner interface {
 	Anonymous(command *CLICommand, willAuthenticate bool) func(*cobra.Command, []string) error
 	Authenticated(command *AuthenticatedCLICommand) func(*cobra.Command, []string) error
+	AuthenticatedWithAPIKey(command *AuthenticatedCLICommand) func(*cobra.Command, []string) error
```

Comment:
Can we combine this with `Authenticated()` so the logic to determine if the function should be authenticated with a token or API key lives here instead of duplicated across multiple command declarations?